thorlcr/graph/thgraph.hpp

```diff
@@ -1092,7 +1092,7 @@ class graph_decl CActivityBase : implements CInterfaceOf<IThorRowInterfaces>, im
     bool timeActivities; // purely for access efficiency
     bool receiving, cancelledReceive, initialized, reInit;
     Owned<IThorGraphResults> ownedResults; // NB: probably only to be used by loop results
-    CRuntimeStatisticCollection stats;
+    CRuntimeStatisticCollection stats; // NB: these are current live act. stats - merged with inactive via gatherActiveStats
```

Comment:
I think it should be possible/desirable to delete this, and only have inactive.

Reply:
I agree.. I took that approach initially.
There are quite a lot of updates to 'stats' at the moment, but I think they could all be to inactiveStats, since they are overwritten each time they are updated.

But I backed out of that change, because of the way 'stats' is passed into the code context and used by registerTimer.
I suspect the code context object should instead have it's own CRuntimeStatisticCollection , and during activity.serializeStats for it's stats to be requested and merged

Reply:
@ghalliday - pushed another commit (3rd) - that removes the dependency on the activities 'stats' from the code context, and changes it so it has it's own collection which is gathered on serialization.
Please take a look/comment.